algorithms/src/Kokkos_Random.hpp

```diff
@@ -887,18 +896,20 @@ class Random_XorShift64 {
 
 template <class DeviceType = Kokkos::DefaultExecutionSpace>
 class Random_XorShift64_Pool {
+ public:
+  using device_type = typename DeviceType::device_type;
+
  private:
-  using execution_space = typename DeviceType::execution_space;
-  using locks_type      = View<int**, DeviceType>;
-  using state_data_type = View<uint64_t**, DeviceType>;
+  using execution_space = typename device_type::execution_space;
+  using locks_type      = View<int**, device_type>;
+  using state_data_type = View<uint64_t**, device_type>;
   locks_type locks_;
   state_data_type state_;
   int num_states_;
   int padding_;
 
  public:
   using generator_type = Random_XorShift64<DeviceType>;
-  using device_type    = DeviceType;
```

Comment:
Undo.  It bloats the diff and makes us interleave public and private members with no obvious improvement.

Reply:
The improvement is that we are sure that `device_type` is a device type in fact.

Reply:
Oh you are right I missed that